tests/exposures/test_reinsurance.py

```diff
@@ -0,0 +1,227 @@
+                 'loc_1_2_PR',
+                 'loc_PR',
+                 'pol_PR',
+                 'simple_loc_FAC']
+
+
+fm_examples = ['fm24',
+               'fm27']
+
+
+test_cases = []
+for case in test_examples + fm_examples:
+    test_cases.append((
+        os.path.join(input_dir, case),
+        os.path.join(expected_output_dir, case)
+    ))
+
+
+class TestReinsurance(unittest.TestCase):
+
+    def _run_fm(
+            self,
+            input_name,
+            output_name,
+            xref_descriptions,
+            allocation=oed.ALLOCATE_TO_ITEMS_BY_PREVIOUS_LEVEL_ALLOC_ID):
+        command = "fmcalc -p {0} -n -a {2} < {1}.bin | tee {0}.bin | fmtocsv > {0}.csv".format(
+            output_name, input_name, allocation)
+        print(command)
+        proc = subprocess.Popen(command, shell=True)
+        proc.wait()
+        if proc.returncode != 0:
+            raise Exception("Failed to run fm")
+        losses_df = pd.read_csv("{}.csv".format(output_name))
+        inputs_df = pd.read_csv("{}.csv".format(input_name))
+
+        losses_df.drop(losses_df[losses_df.sidx != 1].index, inplace=True)
+        inputs_df.drop(inputs_df[inputs_df.sidx != 1].index, inplace=True)
+        losses_df = pd.merge(
+            inputs_df,
+            losses_df, left_on='output_id', right_on='output_id',
+            suffixes=('_pre', '_net'))
+
+        losses_df = pd.merge(
+            xref_descriptions,
+            losses_df, left_on='xref_id', right_on='output_id')
+
+        del losses_df['event_id_pre']
+        del losses_df['sidx_pre']
+        del losses_df['event_id_net']
+        del losses_df['sidx_net']
+        del losses_df['output_id']
+        del losses_df['xref_id']
+        return losses_df
+
+    def _run_test(
+            self,
+            account_df, location_df, ri_info_df, ri_scope_df,
+            loss_factor,
+            do_reinsurance):
+        """
+        Run the direct and reinsurance layers through the Oasis FM.
+        Returns an array of net loss data frames, the first for the direct layers
+        and then one per inuring layer.
+        """
+        t_start = time.time()
+
+        net_losses = OrderedDict()
+
+        initial_dir = os.getcwd()
+        try:
+
+            with TemporaryDirectory() as run_dir:
+
+                os.chdir(run_dir)
+
+                direct_layer = DirectLayer(account_df, location_df)
+                direct_layer.generate_oasis_structures()
+                direct_layer.write_oasis_files()
+                losses_df = direct_layer.apply_fm(
+                    loss_percentage_of_tiv=loss_factor, net=False)
+                net_losses['Direct'] = losses_df
+
+                oed_validator = oed.OedValidator()
+                if do_reinsurance:
+                    (is_valid, error_msgs) = oed_validator.validate(ri_info_df, ri_scope_df)
+                    if not is_valid:
+                        print(error_msgs)
+                        exit(1)
```

Comment:
@mpinkerton-oasis Using `exit` here will result in the test process itself being stopped - the preferred outcome would be to skip the test.

Another problem with these tests is that in order for them to work the working directory must be `/path/to/OasisLMF/tests/exposures` - you are setting this via `os.cwd()`. If you ran the test module from `/path/to/OasisLMF` the tests all fail with that system exit, because the expected test output directory would be a subfolder of `/path/to/OasisLMF/`, e.g. `/path/to/OasisLMF/examples` etc.

The structure of this test module, including the hard coded test data and how it is used in the module, causes problems with running pytest and tox in the normal way, which is to run from the base of the repository.

I was going to merge in this branch to the FM branch, but because of these problems, which would prevent tox from passing all the other tests, I will leave it out.